dali/test/python/operator_2/test_warp.py

```diff
@@ -190,79 +190,41 @@ def compare(pipe1, pipe2, max_err):
-                      " matrix as input: ", use_input,
-                      " input_type: ", itype,
-                      " output_type: ", otype,
-                      " map_inverse:", inv_map)
-                cv_pipeline = CVPipeline(batch_size, otype, itype, use_input, inv_map=inv_map)
-                cv_pipeline.build()
-
-                cpu_pipeline = WarpPipeline("cpu", batch_size, otype, itype, use_input,
-                                            inv_map=inv_map)
-                cpu_pipeline.build()
-
-                compare(cv_pipeline, cpu_pipeline, 8)
-
-
-def test_gpu_vs_cv():
-    random.seed(1007)
-    for batch_size in [1, 4, 19]:
+    for device in ['cpu', 'gpu']:
         for use_input in [False, True]:
             for (itype, otype) in io_types:
                 inv_map = random.choice([False, True])
-                print("Testing gpu vs cv",
-                      "\nbatch size: ", batch_size,
-                      " matrix as input: ", use_input,
-                      " input_type: ", itype,
-                      " output_type: ", otype,
-                      " map_inverse:", inv_map)
-                cv_pipeline = CVPipeline(batch_size, otype, itype, use_input, inv_map=inv_map)
-                cv_pipeline.build()
-
-                gpu_pipeline = WarpPipeline("gpu", batch_size, otype, itype, use_input,
-                                            inv_map=inv_map)
-                gpu_pipeline.build()
-
-                compare(cv_pipeline, gpu_pipeline, 8)
+                batch_size = random.choice([1, 4, 19])
+                yield impl, device, batch_size, use_input, otype, itype, inv_map
 
 
 def test_gpu_vs_cpu():
-    random.seed(1005)
-    for batch_size in [1, 4, 19]:
-        for use_input in [False, True]:
-            for (itype, otype) in io_types:
-                inv_map = random.choice([False, True])
-                print("Testing gpu vs cpu",
-                      "\nbatch size: ", batch_size,
-                      " matrix as input: ", use_input,
-                      " input_type: ", itype,
-                      " output_type: ", otype,
-                      " map_inverse:", inv_map)
-                cpu_pipeline = WarpPipeline("cpu", batch_size, otype, itype, use_input,
-                                            inv_map=inv_map)
-                cpu_pipeline.build()
-
-                gpu_pipeline = WarpPipeline("gpu", batch_size, otype, itype, use_input,
-                                            inv_map=inv_map)
-                gpu_pipeline.build()
-
-                compare(cpu_pipeline, gpu_pipeline, 1.0001)
+    def impl(batch_size, use_input, otype, itype, inv_map):
+        cpu_pipeline = WarpPipeline("cpu", batch_size, otype, itype, use_input, inv_map=inv_map)
+        cpu_pipeline.build()
+        gpu_pipeline = WarpPipeline("gpu", batch_size, otype, itype, use_input, inv_map=inv_map)
+        gpu_pipeline.build()
 
+    random.seed(1005)
```

Comment:
Make sure to pick a good seed - I tried that and with 1005 the distribution wasn't good.